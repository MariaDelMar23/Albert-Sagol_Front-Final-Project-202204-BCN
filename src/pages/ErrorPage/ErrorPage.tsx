import Error404 from "../../components/Error404/Error404";
import Header from "../../components/Header/Header";
import NavigationBar from "../../components/NavigationBar/NavigationBar";

const ErrorPage = (): JSX.Element => {
  return (
    <>
      <Header pageName="Error" adminUser={false} />
      <NavigationBar />
      <Error404 />
    </>
  );
};

export default ErrorPage;
