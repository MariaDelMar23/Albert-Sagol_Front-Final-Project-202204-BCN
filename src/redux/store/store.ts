import { configureStore } from "@reduxjs/toolkit";
import petsReducer from "../features/petsSlice";
import userReducer from "../features/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    pets: petsReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
