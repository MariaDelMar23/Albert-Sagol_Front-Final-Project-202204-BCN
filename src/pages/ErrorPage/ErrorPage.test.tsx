import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../redux/store/store";
import ErrorPage from "./ErrorPage";

describe("Given the ErrorPage component", () => {
  describe("When it's called", () => {
    test("Then it should render a heading with 'Error'", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <ErrorPage />
          </Provider>
        </BrowserRouter>
      );

      expect(
        screen.getByRole("heading", { level: 1, name: "Error" })
      ).toBeInTheDocument();
    });
  });
});
