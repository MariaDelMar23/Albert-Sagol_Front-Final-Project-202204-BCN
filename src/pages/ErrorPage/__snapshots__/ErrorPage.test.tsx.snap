// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ErrorPage component When it's called Then it should match with the snapshot 1`] = `
Array [
  <header
    className="sc-gsnTZi jxSYoa"
  >
    <div
      className="container--header-logo-title"
    >
      <img
        alt="Pet's House logo"
        src="./images/logo.svg"
      />
    </div>
    <h1>
      Error
    </h1>
    <div
      className="container--header-buttons"
    />
  </header>,
  <div
    className="sc-dkzDqf ivpBhO"
  >
    <div
      className="nav__container--list"
    >
      <img
        alt="home icon"
        onClick={[Function]}
        src="./images/icon-home.svg"
      />
      <span>
        List
      </span>
    </div>
    <div
      className="nav__container--logout"
    >
      <img
        alt="logout icon"
        onClick={[Function]}
        src="./images/icon-logout.svg"
      />
      <span>
        Logout
      </span>
    </div>
  </div>,
  <main
    className="sc-bczRLJ cSJOOb"
  >
    <div>
      <span>
        404 not found
      </span>
    </div>
  </main>,
]
`;
